doc/source/whatsnew/v1.3.0.rst

```diff
@@ -640,12 +642,11 @@ See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for mor
 
 Other API changes
 ^^^^^^^^^^^^^^^^^
-- Partially initialized :class:`CategoricalDtype` (i.e. those with ``categories=None`` objects will no longer compare as equal to fully initialized dtype objects.
-- Accessing ``_constructor_expanddim`` on a :class:`DataFrame` and ``_constructor_sliced`` on a :class:`Series` now raise an ``AttributeError``. Previously a ``NotImplementedError`` was raised (:issue:`38782`)
```

Comment:
I've removed this note as I think it only pertains to the internals - though I understand downstream packages are using it. Perhaps this is a case where we'd like it in the whatsnew - cc @topper-123

Reply:
I consider this public for developers, who subclass dataframes and series. See [here](https://pandas.pydata.org/docs/development/extending.html#subclassing-pandas-data-structures), where the attributes were (publically) explained.

However this already raised previously, and this PR just changes the exception type, so it's probably not that important. I'd prefer to keep it, but am ok with deleting.

Reply:
It's correct that `_constructor_expanddim` is "public" developer API.

Reply:
Thanks @topper-123 and @jorisvandenbossche - this note has been restored.